x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -1380,6 +1382,33 @@ public void cleanUpFeature(
             unsetResetModeListener::onFailure
         );
 
+        // Remove all trained model aliases
+        ActionListener<AcknowledgedResponse> pipelineValidation = ActionListener.wrap(
+            acknowledgedResponse -> clusterService.submitStateUpdateTask(
+                "reset-model-aliases",
+                // TODO: it would be nice if these timeouts could be customized by the caller of the feature reset API
+                new AckedClusterStateUpdateTask(new AckedRequest() {
+                    @Override
+                    public TimeValue ackTimeout() {
+                        return AcknowledgedRequest.DEFAULT_ACK_TIMEOUT;
+                    }
+
+                    @Override
+                    public TimeValue masterNodeTimeout() {
+                        return AcknowledgedRequest.DEFAULT_MASTER_NODE_TIMEOUT;
+                    }
+                }, afterTrainedModelAliasesRemoved) {
+                    @Override
+                    public ClusterState execute(final ClusterState currentState) {
+                        final ClusterState.Builder builder = ClusterState.builder(currentState);
+                        builder.metadata(Metadata.builder(currentState.getMetadata())
+                            .putCustom(ModelAliasMetadata.NAME, ModelAliasMetadata.EMPTY).build());
+                        return builder.build();
```

Comment:
I wonder, could we simply delete the entry for `ModelAliasMetadata.NAME`? I am thinking of the scenario where a user hasn't ever created a model alias (and consequently, the metadata hasn't even been added to the cluster state yet). This reset call will actually be the first thing to create the metadata entry.